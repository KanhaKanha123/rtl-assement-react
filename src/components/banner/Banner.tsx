import React from 'react';
import { ArticleProps } from "../../types/types";

const Banner = ({ artcleData: { title, label, description, image } }: ArticleProps) => {
    return (<>
        <div data-testid="banner-wrapper-div" className="banner-wrapper">
            <img data-testid="banner-image" src={image}></img>
            <span data-testid="banner-label" className="title">{label}</span>
            <h1 data-testid="banner-title" className="bannerTitle">{title}</h1>
            <span data-testid="banner-description">{description}</span>
        </div>
    </>
    )
}

export default Banner;