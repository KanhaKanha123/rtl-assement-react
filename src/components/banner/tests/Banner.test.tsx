import Banner from '../Banner';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

const testData= {
title: "De uitschieters van 2018",
label: "best gelezen",
description: "Waar hadden we.",
image: "https://www.rtlboulevard.nl/sites/default/files/content/images/2018/12/27/thumbnail_Jaaroverzicht%20Banner.jpg"
};

describe('Banner component test', () => {
    beforeEach(() => {
        render(<Banner artcleData={testData}/>);
    });

    test("Banner component load correctly", () => {
        const wrapper=screen.getByTestId("banner-wrapper-div");
        const bannerImage=screen.getByTestId("banner-image");
        const bannerLabel=screen.getByTestId("banner-label");
        const bannerBitle=screen.getByTestId("banner-title");
        const bannerDescription=screen.getByTestId("banner-description")
       
        //check element exist
        expect(wrapper).toBeInTheDocument();
        expect(bannerImage).toBeInTheDocument();
        expect(bannerLabel).toBeInTheDocument();
        expect(bannerBitle).toBeInTheDocument();
        expect(bannerDescription).toBeInTheDocument();
    });

    test("Banner component recieve correct props", () => {
        const bannerLabel=screen.getByTestId("banner-label");
        const bannerBitle=screen.getByTestId("banner-title");
        const bannerDescription=screen.getByTestId("banner-description")

        //check values
        expect(bannerDescription).toHaveTextContent('Waar hadden we.');
        expect(bannerLabel).toHaveTextContent('best gelezen');
        expect(bannerBitle).toHaveTextContent('De uitschieters van 2018');
    });

    test("Banner component should match snapshot", () => {
        const element = renderer.create(<Banner artcleData={testData} />);
        expect(element.toJSON()).toMatchSnapshot();
    });
})
