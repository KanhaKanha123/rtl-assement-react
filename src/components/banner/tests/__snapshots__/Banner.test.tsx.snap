// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component test Banner component should match snapshot 1`] = `
<div
  className="banner-wrapper"
  data-testid="banner-wrapper-div"
>
  <img
    data-testid="banner-image"
    src="https://www.rtlboulevard.nl/sites/default/files/content/images/2018/12/27/thumbnail_Jaaroverzicht%20Banner.jpg"
  />
  <span
    className="title"
    data-testid="banner-label"
  >
    best gelezen
  </span>
  <h1
    className="bannerTitle"
    data-testid="banner-title"
  >
    De uitschieters van 2018
  </h1>
  <span
    data-testid="banner-description"
  >
    Waar hadden we.
  </span>
</div>
`;
